services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -0,0 +1,180 @@
+import org.openkilda.model.SwitchId
+import org.openkilda.northbound.dto.v2.switches.PortHistoryResponse
+import org.openkilda.testing.service.northbound.NorthboundServiceV2
+
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+@Narrative("Verify that port history is created for the port up/down actions.")
+class PortHistorySpec extends HealthCheckSpecification {
+    @Autowired
+    NorthboundServiceV2 northboundV2
+
+    @Unroll
+    def "Port history are created for the port down/up actions when link is #islDescription"() {
+        given: "A link"
+        assumeTrue("Unable to find $islDescription ISL for this test", isl as boolean)
+        def timestampBefore = System.currentTimeSeconds()
+
+        when: "Execute port DOWN on the src switch"
+        northbound.portDown(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "Port is really DOWN"
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LINK_DOWN"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.FAILED
+        }
+
+        and: "Port history is created on the src switch"
+        Long timestampAfterDown = System.currentTimeSeconds()
+        with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterDown)) {
+            it.size() == 1
+            checkHistoryPortDownAction(it[0], isl.srcSwitch.dpId, isl.srcPort)
+        }
+
+        when: "Execute port UP on the src switch"
+        northbound.portUp(isl.srcSwitch.dpId, isl.srcPort)
+
+        then: "Port is really UP"
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getPort(isl.srcSwitch.dpId, isl.srcPort).state[0] == "LIVE"
+            assert islUtils.getIslInfo(isl).get().state == IslChangeType.DISCOVERED
+        }
+
+        and: "Port history is updated on the src switch"
+        Long timestampAfterUp = System.currentTimeSeconds()
+        with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterUp)) {
+            it.size() == 2
+            checkHistoryPortUpAction(it[0], isl.srcSwitch.dpId, isl.srcPort)
+        }
+
+        and: "Port history is not empty when link is direct"
+        northboundV2.getPortHistory(isl.dstSwitch.dpId, isl.dstPort, timestampBefore, timestampAfterUp).size() ==
+                historySizeOnDstSw
+
+        and: "Port history is also available using default timeline"
+        northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort).size() >= 2
+
+        where:
+        [islDescription, historySizeOnDstSw, isl] << [
+                ["direct", 2, getTopology().islsForActiveSwitches.find { !it.aswitch }],
+                ["a-switch", 0, getTopology().islsForActiveSwitches.find {
+                    it.aswitch?.inPort && it.aswitch?.outPort
+                }]
+        ]
+    }
+
+    @Tags(SMOKE)
```

Comment:
Not interesting for SMOKE, since it has low value. I find the above test to be more valuable and important than this one. SMOKE should be more about happy paths, negative cases usually have lower priority

Reply:
this test was chosen because we can test positive and negative behaviour at the same time.
it was fixed, new test was marked as SMOKE
```
 @IterationTag(tags = [SMOKE], iterationNameRegex = /direct/)
    def "Port history are created for the port down/up actions when link is #islDescription"() {
```

Reply:
I didn't think of it this way, thought it was just about checking that the port history is empty for wrong period. It's ok now anyway